packages/typespec-client-generator-core/design-docs/flags.md

```diff
@@ -0,0 +1,35 @@
+# TCGC `tspconfig.yaml` flags
+
+## 1. `generate-protocol-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@protocolAPI` is not set for an operation. Default value is `true`.
+
+## 2. `generate-convenience-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@convenientAPI` is not set for an operation. Default value is `true`.
+
+## 3. `package-name`
+
+The `package-name` override you would like for your generated library. By default individual language emitters should calculate the package name for their library from the `SdkPackage.namespaces` that `tcgc` returns to them.
+
+Currently, emitters don't really use it, but it would be great to move language emitters to use this flag. Java should be excluded from this list because it doesn't apply to their naming structure.
+
+Should remove any logic that derives namespace names from the value of `package-name`.
+
+Default value is `none`.
+
+## 4. `flatten-union-as-enum`
+
+Whether you want TCGC to return unions of strings as enums. Default value is `true`. I feel this should not be a `tspconfig.yaml` flag, but an option you can pass to TCGC when calling `createSdkContext`
```

Comment:
How do we configure it if it's not a flag?

Reply:
it can be passed in as an option to `createSdkContext` which is called by the individual emitters. So emitters would look like `createSdkContext(..., {flattenUnionAsEnum: false})`

Reply:
That doesnt' reply to my question :). We made it a flag so users can decide the behavior they want, how to user decide what behavior they want, if it's not through a flag?

Reply:
it is not a flag for end user. it is an option for emitter. let me refine the doc.

Reply:
it should not configured by user. it is only used by emitter. i moved it from flag to options.